### ISSUE TITLE
Inconsistent accessibility of function return type

### ISSUE THREAD

**Issue by mattmc3** [boo-dotnet/boo-codehaus-archive#1341]
| Field | Value |
|---|---|
| Type | Bug |
| Status | Open |
| Priority | Major |
| Resolution | Unresolved |
| Reporter | Maksym Trushyn |
| Assignee | Unassigned |
| Affects | 0.9.4 |
| Fix for | None |
| Components | Compiler |
| Created | 2012-02-14T21:00-0600 |
| Updated | 2012-03-05T14:00-0600 |

## Description

I found possible issue while working with method Boo.OMeta.OMetaInput.Prepend

The method has inferred type of OMetaInputCons. Type OMetaInputCons is internal. But method Prepend is public. 

I think that error should be returned in this case. C# returns the following error message in this case "Inconsistent accessibility: return type ' Boo.OMeta.OMetaInputCons' is less accessible than method 'Boo.OMeta.OMetaInput.Prepend".

## Comments (1)

---

### Maksym Trushyn, 05/Mar/12 2:00 PM

Just adding the simpler scenario.

In the example provided below the static method A.GetA() is of inferred type B. I think it is an issue because type B is less accessible than method A.GetA(). The problem is undetectable during compile time. I found the issue when I placed the class C to another assembly and call method C.GetA(). The following run-time exception was raised: System.TypeAccessException : Attempt by method 'C.GetA()' to access type 'B' failed.

import System

class A:

 static def GetA():

 return B()

internal class B(A):

 pass

class C:

 def GetA():

 a = A.GetA()

a = A.GetA()

assert not a isa B

assert a isa A 

print "Press any key to continue . . . "

Console.ReadKey(true)

## Provenance

- <http://web.archive.org/web/20150520081122id_/https://jira.codehaus.org/browse/BOO-1341>
- <http://web.archive.org/web/20150521151729id_/https://jira.codehaus.org/browse/BOO-1341?page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel>
